src/frontend/screens/AppPasscode/index.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { defineMessages, FormattedMessage } from "react-intl";
+import { StyleSheet, View } from "react-native";
+import { useNavigation } from "react-navigation-hooks";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import SettingsContext from "../../context/SettingsContext";
+import HeaderTitle from "../../sharedComponents/HeaderTitle";
+import IconButton from "../../sharedComponents/IconButton";
+import { BackIcon } from "../../sharedComponents/icons";
+import { SecurityContext } from "../Security/SecurityContext";
+import { EnterPasscode } from "./EnterPasscode";
+import { NewPasscode } from "./NewPasscode";
+
+const m = defineMessages({
+  title: {
+    id: "screens.AppPasscode",
+    defaultMessage: "App Passcode",
+  },
+});
+
+export const AppPasscode: NavigationStackScreenComponent = () => {
+  const { passIsSet } = React.useContext(SecurityContext);
+  const [{ experiments }] = React.useContext(SettingsContext);
+  const { navigate } = useNavigation();
+
+  React.useEffect(() => {
+    if (!experiments.appPasscode) navigate("Settings");
+  }, []);
```

Comment:
`experiments` should be added to the deps array here

Reply:
technically `navigate` should too but i know that there are some issues with adding that in a use effect (i.e. causing a ton of re-runs), so okay to omit that one here